awx/ui_next/src/App.jsx

```diff
@@ -39,6 +46,48 @@ function App() {
   const match = useRouteMatch();
   const { hash, search, pathname } = useLocation();
 
+  const fetchUserData = async () => {
```

Comment:
This adds a lot of logic to the App component... I'm wondering if it can be moved down into a new component instead (maybe something wrapping `<Login>`? We did a lot of work last year to keep these top level components nice and streamlined, and I'd hate to lose some of that

Reply:
hey @keithjgrant, thanks for your comment. I did think about a context that could be updated by the login component, and store information about the logged user. This could avoid de-duplicate a few requests inside the AppContainer that check the same data. The drawback is that we will have one of more context. In this case we will probably need to wrap the whole application on this new context.  I will explore the idea to move logic to the login component. If you have a specific suggestion, just let me know.

Reply:
Yeah, I'd avoid a new context I think. I'd see what would happen you either move it into tho Login component, or create a new component that wraps the Login component to encapsulate that logic and expose the callbacks props needed to interact with it